src/main/java/org/stellar/sdk/MemoText.java

```diff
@@ -23,7 +23,7 @@ public MemoText(XdrString text) {
     this.text = checkNotNull(text, "text cannot be null");
     int length = this.text.getBytes().length;
     if (length > 28) {
-      throw new MemoTooLongException("text must be <= 28 bytes. length=" + String.valueOf(length));
+      throw new MemoWrongSizeException("text must be <= 28 bytes. length=" + String.valueOf(length), length);
```

Comment:
We should keep the too long exception for memo text. Otherwise, this breaks backwards compatibility for code that is outside the area affected by the bug we are fixing.